meta-client/src/main/java/org/apache/helix/metaclient/factories/MetaClientConfig.java

```diff
@@ -19,23 +19,32 @@
  * under the License.
  */
 
-class MetaClientConfig {
+import org.apache.helix.metaclient.constants.MetaClientConstants;
+
+public class MetaClientConfig {
 
   public enum StoreType {
     ZOOKEEPER, ETCD, CUSTOMIZED
   }
 
   private final String _connectionAddress;
-  private final long _connectionTimeout;
+
+  // Wait for init timeout time until connection is initiated
+  private final long _connectionInitTimeoutInMillis;
+
+  // When a client becomes partitioned from the metadata service for more than session timeout,
+  // new session will be established when reconnect.
+  private final long _sessionTimeoutInMillis;
+
   private final boolean _enableAuth;
```

Comment:
Should we add this now? I think we should add whole security APIs and config stuff later a single PR because just _enableAuth won't be enough but whole "AuthConfig" objection should be passed to builder class. This AuthConfig will have lot of different configs in it.